### PR TITLE
Add filemanager plugin

### PR DESCRIPTION
**Pull request by 0mkara**
- [x] vscode specific native filemanger plugin

### CODE REVIEW

File: packages/engine/vscode/src/lib/filemanager.ts [hunk 1]
```diff
@@ -0,0 +1,150 @@
...
+ * `export default class FileManagerPlugin extends CommandPlugin implements NextFileSystemApi`
+ */
+
+export default class FileManagerPlugin extends CommandPlugin {
+  constructor() {
+    super(profile);
+  }
+  /** Open the content of the file in the context (eg: Editor) */
+  open(path: string): Promise<boolean> {
+    return new Promise((resolve, reject) => {
```
Comment (GrandSchtroumpf) [inline packages/engine/vscode/src/lib/filemanager.ts:44]: You don't need to create a promise here : 
```typescript
open(path: string): Promise<boolean> {
  const uri = Uri.file(path);
  commands.executeCommand('vscode.open', uri).then(() => true);
}
```

Comment (GrandSchtroumpf) [reply packages/engine/vscode/src/lib/filemanager.ts:44]: This apply for all below

File: packages/engine/vscode/src/lib/filemanager.ts [hunk 2]
```diff
@@ -0,0 +1,150 @@
...
+        .then(() => {
+          resolve(true);
+        })
+    })
+  }
+  /** Set the content of a specific file */
+  writeFile(path: string, data: string): Promise<void> {
+    return new Promise((resolve, reject) => {
+      const turi = Uri.parse(path);
+      const encoder = new TextEncoder();
```
Comment (GrandSchtroumpf) [inline packages/engine/vscode/src/lib/filemanager.ts:56]: Where does the TextEncoder comes from ?

File: packages/engine/vscode/src/lib/filemanager.ts [hunk 3]
```diff
@@ -0,0 +1,150 @@
...
+      const uint8Array = encoder.encode(data);
+      workspace.fs.writeFile(turi, Uint8Array.from(uint8Array)).then(() => {
+        resolve();
+      });
+    });
+  }
+  /** Return the content of a specific file */
+  readFile(path: string): Promise<String> {
```
Comment (GrandSchtroumpf) [inline packages/engine/vscode/src/lib/filemanager.ts:64]: Just a little improvement : 
```typescript
readFile(path: string): Promise<String> {
  const uri = Uri.parse(path);
  return workspace.fs.readFile(uri).then(content => Buffer.from(content).toString("utf-8"))
}
```

File: packages/engine/vscode/src/lib/filemanager.ts [hunk 4]
```diff
@@ -0,0 +1,150 @@
...
+      const duri = Uri.parse(path);
+      workspace.fs.readDirectory(duri).then((data) => {
+        resolve(data);
+      });
+    });
+  }
+  // ------------------------------------------
+  // Legacy API. To be removed.
+  // ------------------------------------------
+  async getFolder(path: string): Promise<Folder> {
```
Comment (GrandSchtroumpf) [inline packages/engine/vscode/src/lib/filemanager.ts:116]: It should only be a redirect to `readdir` : 
```typescript
getFolder = readdir;
getFile = readFile;
setFile = writeFile;
switchFile = open;
getCurrentFile() {
  return window.activeTextEditor?.document?.fileName;
}
```

**Review by GrandSchtroumpf — CHANGES_REQUESTED** [review]

File: packages/engine/vscode/src/lib/filemanager.ts (hunk 2 above)
Comment (0mkara) [reply packages/engine/vscode/src/lib/filemanager.ts:56]: From typescript.

ref: https://developer.mozilla.org/en-US/docs/Web/API/TextEncoder

File: packages/engine/vscode/src/lib/types.ts [hunk 5]
```diff
@@ -0,0 +1,5 @@
+export interface Folder {
```
Comment (GrandSchtroumpf) [inline packages/engine/vscode/src/lib/types.ts:1]: Normally you don't need that anymore

**Review by GrandSchtroumpf — APPROVED** [review]

**Review by GrandSchtroumpf — APPROVED** [review]

**Pull request merged**